async_tcp_server.hpp

```diff
@@ -0,0 +1,28 @@
+#ifndef ASYNC_TCP_SERVER
+#define ASYNC_TCP_SERVER
+
+#include "socket_wrapper.hpp"
+#include "async_engine.hpp"
+#include <vector>
+
+using Socket_vector = std::vector<Socket_sptr>;
+
+class AsyncTCPServer
+{
+
+    AsyncEngine_uptr                m_async_engine;
+
+    public:
+
+                                    AsyncTCPServer();
+                                    ~AsyncTCPServer();
+    void                            SendEcho(Socket_sptr, const char*, const int);
+    void                            RunAsyncServer(const int server_port);
+    Socket_sptr                     AcceptIncomingConnection();
+    void                            CloseConnection(Socket_sptr);
+    bool                            ShouldConnectClient(Socket_sptr);
```

Comment:
Do we really need all those functions to be `public`?
I'm almost sure the only API this class should expose is `Run`, at least in this specific case.